src/components/DataSession/DataSession.vue

```diff
@@ -57,24 +58,42 @@ function addCompletedOperation(operationResponse) {
       outputFile.operationIndex = operationMap[operationResponse.id]
       if (!imagesContainsFile(outputFile)) {
         images.value.push(outputFile)
-        if (selectedOperation == -1 || selectedOperation == outputFile.operationIndex) {
+        if (selectedOperation == -1 || selectedOperation == outputFile.operation) {
           filteredImages.value.push(outputFile)
         }
       }
     })
   }
 }
 
-
 function selectOperation(operationIndex) {
-  if (operationIndex != selectedOperation) {
-    selectedOperation = operationIndex
-    if (selectedOperation == -1) {
-      filteredImages.value = images.value
+  if (operationIndex != operationMap[selectedOperation]) {
+    if (operationIndex == -1) {
+      selectedOperation = -1
     }
     else {
-      filteredImages.value = images.value.filter(image => image.operationIndex == operationIndex)
+      selectedOperation = props.data.operations[operationIndex].id
     }
+    reconcileFilteredImages()
+  }
+}
+
+function reconcileOperationImages() {
```

Comment:
could you add some comments to explain why we're doing this? Just to help me/anyone who works on it in the future